xc/common/cmake/vivado.cmake

```diff
@@ -127,7 +127,7 @@ function(COMMON_VIVADO_TARGETS)
   add_custom_target(${NAME}_timing DEPENDS ${WORK_DIR}/timing_${NAME}.json)
   add_custom_target(${NAME}_fasm DEPENDS ${WORK_DIR}/design_${NAME}.bit.fasm)
 
-  if(${COMMON_VIVADO_TARGETS_MAKE_DIFF_FASM})
+  if(${COMMON_VIVADO_TARGETS_CHECK_DIFF_FASM})
```

Comment:
Change the leaf target purpose doesn't seem like the right thing to do?  Rather than modify what the `_diff_fasm` target does, we should change what the `all_x_diff_fasm` depends on.  On line 128 is `${NAME}_fasm`, which just needs the `${DEPS}` added.  At the next level up, we can tie the `${NAME}_fasm` target to the `all` target to achieve the affect we want.

Reply:
Ok, I have added a cached variable to control whether the Verilog/XDC targets should link the `{NAME}_fasm` or the `{NAME}_diff_fasm` to the `all_{}_diff_fasm` target